service-model-generator/modules/service-model-generator-ls-extension/src/main/java/io/ballerina/servicemodelgenerator/extension/ServiceModelGeneratorService.java

```diff
@@ -1114,6 +1117,27 @@ public CompletableFuture<CommonSourceResponse> updateClassField(ClassFieldModifi
         });
     }
 
+    /**
+     * Get the filtered list of types for a given protocol context.
+     *
+     * @param request Class field modifier request
+     * @return {@link CommonSourceResponse} of the common source response
+     */
+    @JsonRequest
+    public CompletableFuture<TypeResponse> types(TypesRequest request) {
+        return CompletableFuture.supplyAsync(() -> {
+            TypeResponse response = new TypeResponse(new ArrayList<>());
+            try {
+                Path filePath = Path.of(request.filePath());
+                Project project = this.workspaceManager.loadProject(filePath);
+                TypeCompletionGenerator.getTypes(project, response.completions());
+                return response;
+            } catch (Throwable e) {
+                return response;
+            }
+        });
+    }
```

Comment:
Following is a simpler and better implementation alternative to avoid leaky abstractions
```
CompletableFuture.supplyAsync(() -> {
    try {
        Path filePath = Path.of(request.filePath());
        Project project = this.workspaceManager.loadProject(filePath);

        List<Completion> completions = TypeCompletionGenerator.getTypes(project);
        return new TypeResponse(completions);
    } catch (Throwable e) {
        return new TypeResponse(Collections.emptyList());
    }
});
```